sqlmesh/core/macros.py

```diff
@@ -648,6 +648,87 @@ def eval_(evaluator: MacroEvaluator, condition: exp.Condition) -> t.Any:
     return evaluator.eval_expression(condition)
 
 
+@macro()
+def star(
+    evaluator: MacroEvaluator,
+    relation: exp.Table,
+    alias: t.Optional[exp.Identifier | exp.Column] = None,
+    except_: t.Optional[exp.Array | exp.Tuple] = None,
+    prefix: t.Optional[exp.Literal] = None,
+    suffix: t.Optional[exp.Literal] = None,
+    quote_identifiers: exp.Boolean = exp.true(),
+) -> t.List[exp.Expression]:
+    """Returns a list of projections for the given relation.
+
+    Example:
+        >>> from sqlglot import parse_one
+        >>> from sqlmesh.core.macros import MacroEvaluator
+        >>> sql = "SELECT @STAR(foo, bar, [c], 'baz_') FROM foo AS bar"
+        >>> MacroEvaluator(schema={"foo": {"a": "string", "b": "string", "c": "string", "d": "int"}}).transform(parse_one(sql)).sql()
+        'SELECT CAST("bar"."a" AS TEXT) AS "baz_a", CAST("bar"."b" AS TEXT) AS "baz_b", CAST("bar"."d" AS INT) AS "baz_d" FROM foo AS bar'
+    """
+    if alias and not isinstance(alias, (exp.Identifier, exp.Column)):
+        raise SQLMeshError(f"Invalid alias '{alias}'. Expected an identifier.")
+    if except_ and not isinstance(except_, (exp.Array, exp.Tuple)):
+        raise SQLMeshError(f"Invalid except '{except_}'. Expected an array.")
+    if prefix and not isinstance(prefix, exp.Literal):
+        raise SQLMeshError(f"Invalid prefix '{prefix}'. Expected a literal.")
+    if suffix and not isinstance(suffix, exp.Literal):
+        raise SQLMeshError(f"Invalid suffix '{suffix}'. Expected a literal.")
+    if not isinstance(quote_identifiers, exp.Boolean):
+        raise SQLMeshError(
+            f"Invalid quote_identifiers '{quote_identifiers}'. Expected a boolean."
+        )
+    projections = []
+    exclude = []
+    kwargs = {"quoted": quote_identifiers.this}
+    if alias:
+        kwargs["table"] = alias.name
+    if except_ and isinstance(except_.expressions, list):
```

Comment:
there's no need for this check here, expressions is always a list

Reply:
<img width="736" alt="image" src="https://github.com/TobikoData/sqlmesh/assets/41213451/4349d9ff-b161-430c-873b-3d2dc6936ec0">
 type checker suggests any in unio type but will drop check

Reply:
missing type, i'll fix it